ksm.go

```diff
@@ -0,0 +1,214 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+)
+
+type ksmSetting struct {
+	pagesPerScanFactor uint32
+	scanIntervalMs     uint32
+}
+
+const (
+	ksmSlug       = "slug"
+	ksmSlow       = "slow"
+	ksmStandard   = "standard"
+	ksmAggressive = "aggressive"
+)
+
+var ksmSettings = map[string]ksmSetting{
+	ksmSlug:       {1000, 500}, // Every 500ms, we scan 1 page for every 1000 pages available in the system
```

Comment:
Out of interest - did you have some algo in mind for the increase on the scaling factors and timings?
By my math, we currently have something like (measured in proportion of total system pages scanned per second):
slug: 1/500th of all pages every second
slow: 1/50th
Standard: 1 (all pages)
Aggressive: 100x the number of pages

The aggressive end feels pretty aggressive to me. I might go have a play and see what the cpu cost looks like on an idle system (you can see `ksmd` ticking away in top I think).